### PR TITLE
Add libi2c package

### PR DESCRIPTION
**Pull request by traversaro**
Add the library part of the `i2c-tools` package (see https://i2c.wiki.kernel.org/index.php/I2C_Tools). 

For simplicity, here we package only the library and not i2c command line tools. If people in the future are interested in packaging also the tools (that are licensed under GPL), they can transform the feedstock in a multiple-output feedstock to also create a `i2c-tools`  package with the tools that will depend on `libi2c` .

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge/help-c-cpp The recipe is ready for review.

File: recipes/libi2c/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+make BUILD_DYNAMIC_LIB=1 BUILD_STATIC_LIB=0 install
```
Comment (wolfv) [inline recipes/libi2c/build.sh:3]: does `-j${CPU_COUNT}` work?

Comment (traversaro) [reply recipes/libi2c/build.sh:3]: I am not sure as the `Makefile` of the project is manually written and not generated, I can try. However the package is so small (it builds in like ~15 seconds), so I do not think it plays a big role here.

Comment (traversaro) [reply recipes/libi2c/build.sh]: I am not sure it is actually using more cores, but in any case the compilation works fine.

**Comment by wolfv** [comment]
cool!

**Pull request merged**